qcengine/programs/terachem_pbs.py

```diff
@@ -0,0 +1,103 @@
+"""
+import logging
+import os
+from typing import TYPE_CHECKING
+
+from qcelemental.models import AtomicResult
+from qcelemental.util import which_import
+
+from .model import ProgramHarness
+
+if TYPE_CHECKING:
+    from qcelemental.models import AtomicInput
+
+    from ..config import TaskConfig
+
+logger = logging.getLogger(__name__)
+
+
+class TeraChemPBSHarness(ProgramHarness):
+
+    _defaults = {
+        "name": "terachem_pbs",
+        "scratch": False,
+        "thread_safe": False,
+        "thread_parallel": False,
+        "node_parallel": False,
+        "managed_memory": True,
+    }
+
+    class Config(ProgramHarness.Config):
+        pass
+
+    @staticmethod
+    def found(raise_error: bool = False) -> bool:
+        """Whether TeraChemPBS harness is ready for operation.
+        Parameters
+        ----------
+        raise_error: bool
+            Passed on to control negative return between False and ModuleNotFoundError raised.
+        Returns
+        -------
+        bool
+            If tcpb package is found and server available, returns True.
+            If raise_error is False and tcpb package missing and/or server us unavailable, returns False.
+            If raise_error is True and tcpb package missing and/or server us unavailable, the error message is raised.
+        """
+        tcpb_pkg_available = which_import(
+            "tcpb",
+            return_bool=True,
+            raise_error=raise_error,
+            raise_msg="TeraChem protobuf client package (tcpb) not found. Please install tcpb>=0.7.0.",
+        )
+        if not tcpb_pkg_available:
+            return False
+
+        from tcpb.tcpb import TCProtobufClient
+        from tcpb.exceptions import ServerError
+
+        try:
+            with TCProtobufClient(
+                host=os.getenv("TERACHEM_PBS_HOST"), port=int(os.getenv("TERACHEM_PBS_PORT"))
+            ) as client:
+                return client.is_available()
+        except TypeError as e:
+            # TERACHEM_PBS_HOST/PORT environment variables unset
+            msg = "Environment variables 'TERACHEM_PBS_HOST' and 'TERACHEM_PBS_PORT' must be set!"
+            logger.error(msg)
+            if raise_error:
+                raise ValueError(msg) from e
+        except ServerError as e:
+            msg = (
+                f"Unable to connect to TeraChem server at "
+                f"{os.getenv('TERACHEM_PBS_HOST')}:{os.getenv('TERACHEM_PBS_PORT')}"
+            )
+            logger.error(msg)
+            if raise_error:
+                raise OSError(msg) from e
+        return False
+
+    def get_version(self) -> str:
+        """Returns version of TeraChem Protocol Buffer Server"""
+        try:
+            import tcpb
+        except ModuleNotFoundError:
+            return "???"
+        else:
+            try:
+                return tcpb.__version__
+            except AttributeError:
+                return "???"
```

Comment:
I returned "???" because before I had the `get_version` method I saw this was what `QCEngine` returned by default. So I was sticking to that convention. Should I return something else instead?

Reply:
I'd feel better without those "???" around. Sorry that that poor example was there to be found. I think you can do like https://github.com/MolSSI/QCEngine/blob/master/qcengine/programs/dftd4.py#L47-L57 replacing `dftd4` by `tcpb`